### PR TITLE
tidb-in-kubernetes: Add upgrade tidb documentation

### PR DESCRIPTION
**Pull request by aylei**
Preview: https://github.com/aylei/docs-cn/blob/rolling-update-tidb/tidb-in-kubernetes/upgrade-tidb.md

This documentation has a English version which is not properly organized, I'll commit that to `pingcap/docs` later, namely, this documentation do not need translation😜

@tennix @weekface @lilin90 @dcalvin  PTAL

Signed-off-by: Aylei <rayingecho@gmail.com>

### CODE REVIEW

File: tidb-in-kubernetes/upgrade-tidb.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+title: 滚动更新 TiDB 集群
+category: how-to
+---
+
+# 滚动更新 TiDB 集群
+
+滚动更新 TiDB 集群时，会按 PD / TiKV / TiDB 的顺序，串行删除 Pod，并创建新版本的 Pod，当新版本的 Pod 正常运行后，再处理下一个 Pod。滚动升级过程会自动处理 PD / TiKV 的 Leader 迁移与 TiDB 的 DDL Owner 迁移。因此，在多节点的部署拓扑下（最小环境：PD * 3、TiKV * 3、TiDB * 2），滚动更新不会影响业务运行。
```
Comment (weekface) [inline tidb-in-kubernetes/upgrade-tidb.md]: `滚动更新不会影响业务运行`

 滚动更新 TiDB 的时候是影响业务的

File: tidb-in-kubernetes/upgrade-tidb.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+滚动更新可以用于升级 TiDB 版本，也可以用于更新集群配置。
+
+## 升级 TiDB 版本
+
+1. 修改集群的 `values.yaml` 文件中的 `tidb.image`, `tikv.image`, `pd.image` 的值为新版本镜像；
+2. 执行 `helm upgrade` 命令进行升级：
+
+    ```shell
+    `helm upgrade ${releaseName} charts/tidb-operator -f values.yaml`；
```
Comment (weekface) [inline tidb-in-kubernetes/upgrade-tidb.md]:
```suggestion
    helm upgrade ${releaseName} charts/tidb-cluster -f values.yaml
```

File: tidb-in-kubernetes/upgrade-tidb.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+## 更新 TiDB 集群配置
+
+默认条件下，修改配置文件不会自动应用到 TiDB 集群中，只有在实例重启时，才会重新加载新的配置文件。我们可以开启配置文件自动更新特性，在每次配置文件更新时，自动执行滚动更新，将修改后的配置应用到集群中。操作步骤如下：
+
+1. 修改集群的 `values.yaml` 文件，将 `enableConfigMapRollout` 的值设为 `true`；
+2. 根据需求修改 `values.yaml` 中需要调整的集群配置项；
+3. 执行 `helm upgrade` 命令进行升级：
+
+    ```shell
+    `helm upgrade ${releaseName} charts/tidb-operator -f values.yaml`；
```
Comment (weekface) [inline tidb-in-kubernetes/upgrade-tidb.md]:
```suggestion
    helm upgrade ${releaseName} charts/tidb-cluster -f values.yaml；
```

File: tidb-in-kubernetes/upgrade-tidb.md (hunk 1 above)
Comment (yikeke) [inline tidb-in-kubernetes/upgrade-tidb.md]:
```suggestion
滚动更新 TiDB 集群时，会按 PD/TiKV/TiDB 的顺序，串行删除 Pod，并创建新版本的 Pod，当新版本的 Pod 正常运行后，再处理下一个 Pod。滚动升级过程会自动处理 PD/TiKV 的 Leader 迁移与 TiDB 的 DDL Owner 迁移。因此，在多节点的部署拓扑下（最小环境：PD * 3、TiKV * 3、TiDB * 2），滚动更新不会影响业务运行。
```
No space needed at both sides of "/".

File: tidb-in-kubernetes/upgrade-tidb.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+# 滚动更新 TiDB 集群
+
+滚动更新 TiDB 集群时，会按 PD / TiKV / TiDB 的顺序，串行删除 Pod，并创建新版本的 Pod，当新版本的 Pod 正常运行后，再处理下一个 Pod。滚动升级过程会自动处理 PD / TiKV 的 Leader 迁移与 TiDB 的 DDL Owner 迁移。因此，在多节点的部署拓扑下（最小环境：PD * 3、TiKV * 3、TiDB * 2），滚动更新不会影响业务运行。
+
+滚动更新可以用于升级 TiDB 版本，也可以用于更新集群配置。
+
+## 升级 TiDB 版本
+
+1. 修改集群的 `values.yaml` 文件中的 `tidb.image`, `tikv.image`, `pd.image` 的值为新版本镜像；
```
Comment (yikeke) [inline tidb-in-kubernetes/upgrade-tidb.md]:
```suggestion
1. 修改集群的 `values.yaml` 文件中的 `tidb.image`、`tikv.image`、`pd.image` 的值为新版本镜像；
```
Don't use English commas to list things in a Chinese sentence. Use enumeration commas (顿号) instead.

File: tidb-in-kubernetes/upgrade-tidb.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    watch kubectl -n ${namespace} get pod -o wide
+    ```
+    
+    当所有 Pod 都重建完毕进入 `Running` 状态后，升级完成。
+
+配置文件的滚动更新机制与升级版本相同，不会影响业务正常运行。
+
+> **注意：**
+> 
+> 将 `enableConfigMapRollout` 特性从关闭状态打开时，即使没有配置变更，也会触发一次 PD/TiKV/TiDB 的滚动更新
```
Comment (yikeke) [inline tidb-in-kubernetes/upgrade-tidb.md]:
```suggestion
> 将 `enableConfigMapRollout` 特性从关闭状态打开时，即使没有配置变更，也会触发一次 PD/TiKV/TiDB 的滚动更新。
```

File: tidb-in-kubernetes/upgrade-tidb.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+> **注意：**
+>
+> 目前 PD 的 `scheduler` 和 `replication` 配置(`values.yaml` 中的 `maxStoreDownTime` 和 `maxReplicas` 字段)在集群安装完成后无法自动更新，需要通过 [pd-ctl](/dev/reference/tools/pd-control.md) 手动更新。
```
Comment (yikeke) [inline tidb-in-kubernetes/upgrade-tidb.md]:
```suggestion
> 目前 PD 的 `scheduler` 和 `replication` 配置（`values.yaml` 中的 `maxStoreDownTime` 和 `maxReplicas` 字段）在集群安装完成后无法自动更新，需要通过 [pd-ctl](/dev/reference/tools/pd-control.md) 手动更新。
```

File: tidb-in-kubernetes/upgrade-tidb.md (hunk 5 above)
Comment (yikeke) [inline tidb-in-kubernetes/upgrade-tidb.md]:
```suggestion
> 将 `enableConfigMapRollout` 特性从关闭状态打开时，即使没有配置变更，也会触发一次 PD、TiKV、TiDB 的滚动更新
```

File: tidb-in-kubernetes/upgrade-tidb.md (hunk 4 above)
Comment (aylei) [reply tidb-in-kubernetes/upgrade-tidb.md]: Got it👌

File: tidb-in-kubernetes/upgrade-tidb.md (hunk 2 above)
Comment (aylei) [reply tidb-in-kubernetes/upgrade-tidb.md]: I will change these to the new helm command form (with repo)

File: tidb-in-kubernetes/upgrade-tidb.md (hunk 1 above)
Comment (aylei) [reply tidb-in-kubernetes/upgrade-tidb.md]: ”......滚动更新 TiKV、PD 不会影响业务正常运行。对于有连接重试功能的客户端，滚动更新 TiDB 同样不会影响业务。对于无法进行重试的客户端，滚动更新 TiDB 则会导致连接到被关闭节点的数据库连接失效，造成部分业务请求失败。对于这类业务，推荐在客户端添加重试功能或在低峰期进行 TiDB 的滚动升级操作“

**Comment by aylei** [comment]
@yikeke @weekface All comments are addressed, PTAL again.

File: tidb-in-kubernetes/upgrade-tidb.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+4. 查看升级进度：
+
+    ```shell
+    watch kubectl -n ${namespace} get pod -o wide
+    ```
+    
+    当所有 Pod 都重建完毕进入 `Running` 状态后，升级完成。
+
+配置文件的滚动更新机制与升级版本相同，不会影响业务正常运行。
```
Comment (weekface) [inline tidb-in-kubernetes/upgrade-tidb.md]: 这里是不是也注明下如果客户端没有重试机制会影响业务？

File: tidb-in-kubernetes/upgrade-tidb.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
+---
+title: 滚动更新 TiDB 集群
+category: how-to
+---
+
+# 滚动更新 TiDB 集群
+
+滚动更新 TiDB 集群时，会按 PD、TiKV、TiDB 的顺序，串行删除 Pod，并创建新版本的 Pod，当新版本的 Pod 正常运行后，再处理下一个 Pod。滚动升级过程会自动处理 PD、TiKV 的 Leader 迁移与 TiDB 的 DDL Owner 迁移。因此，在多节点的部署拓扑下（最小环境：PD * 3、TiKV * 3、TiDB * 2），滚动更新 TiKV、PD 不会影响业务正常运行。对于有连接重试功能的客户端，滚动更新 TiDB 同样不会影响业务。对于无法进行重试的客户端，滚动更新 TiDB 则会导致连接到被关闭节点的数据库连接失效，造成部分业务请求失败。对于这类业务，推荐在客户端添加重试功能或在低峰期进行 TiDB 的滚动升级操作。
```
Comment (yikeke) [inline tidb-in-kubernetes/upgrade-tidb.md]:
```suggestion
滚动更新 TiDB 集群时，会按 PD、TiKV、TiDB 的顺序，串行删除 Pod，并创建新版本的 Pod，当新版本的 Pod 正常运行后，再处理下一个 Pod。

滚动升级过程会自动处理 PD、TiKV 的 Leader 迁移与 TiDB 的 DDL Owner 迁移。因此，在多节点的部署拓扑下（最小环境：PD * 3、TiKV * 3、TiDB * 2），滚动更新 TiKV、PD 不会影响业务正常运行。

对于有连接重试功能的客户端，滚动更新 TiDB 同样不会影响业务。对于无法进行重试的客户端，滚动更新 TiDB 则会导致连接到被关闭节点的数据库连接失效，造成部分业务请求失败。对于这类业务，推荐在客户端添加重试功能或在低峰期进行 TiDB 的滚动升级操作。
```
This paragraph is too long. Try to break them apart to improve readability.

File: tidb-in-kubernetes/upgrade-tidb.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+    ```shell
+    watch kubectl -n ${namespace} get pod -o wide
+    ```
+    
+    当所有 Pod 都重建完毕进入 `Running` 状态后，升级完成。
+
+## 更新 TiDB 集群配置
+
+默认条件下，修改配置文件不会自动应用到 TiDB 集群中，只有在实例重启时，才会重新加载新的配置文件。我们可以开启配置文件自动更新特性，在每次配置文件更新时，自动执行滚动更新，将修改后的配置应用到集群中。操作步骤如下：
```
Comment (yikeke) [inline tidb-in-kubernetes/upgrade-tidb.md]:
```suggestion
默认条件下，修改配置文件不会自动应用到 TiDB 集群中，只有在实例重启时，才会重新加载新的配置文件。

您可以开启配置文件自动更新特性，在每次配置文件更新时，自动执行滚动更新，将修改后的配置应用到集群中。操作步骤如下：
```

File: tidb-in-kubernetes/upgrade-tidb.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
+---
```
Comment (yikeke) [inline tidb-in-kubernetes/upgrade-tidb.md]: This file should be put in the docs-cn/how-to/upgrade/ directory. You can refer to [TOC with TiDB in Kubernetes Integrated](https://docs.google.com/document/d/1k7zvGbcoMj1ggo28uZ5gr3-GXNiPAIWda7Df1oXOZbM/edit?usp=sharing) for documentation structure.

**Comment by yikeke** [comment]
Please resolve comments, and we should merge this PR today. `tidb-in-kubernetes` ducuments are expected to be ready before Tech Day on Sunday. @aylei

File: tidb-in-kubernetes/upgrade-tidb.md (hunk 7 above)
Comment (aylei) [reply tidb-in-kubernetes/upgrade-tidb.md]: 是的

Comment (aylei) [reply tidb-in-kubernetes/upgrade-tidb.md]: 这句话我删掉吧，和开头的说明重复了

**Comment by aylei** [comment]
@weekface @yikeke I've addressed the comments, PTAL again 😁

**Pull request merged**